### ISSUE TITLE
[NFR] Guard against positioning above window's top edge

### ISSUE THREAD

**Issue by temuri416** [Flyer53/jsPanel3#52]
Hi,

It would be nice if jsPanel could calculate its top position to see whether it's above window's top edge.

Right now it's possible to open a window that is taller than available height and panel's drag handler is positioned outside of the view, making it impossible to reposition.

Is it possible to prevent top coordinate from going outside top edge?

Thanks!

**Comment by Flyer53** [comment]
Good point! I'm thinking about this once in a while again but did not make a decision yet on how to implement this.

As a workaround I would suggest to check panel left/top values and/or width/height within option.callback and then reposition and/or resize the panel as necessary.

Maybe method resize() is an option because there you can set min/max values for panel width and height.

**Comment by Flyer53** [comment]
@temuri416 
What I totally missed myself is that you have a built-in solution to this already:
The **modify** parameter of option.position http://jspanel.de/api/#option/position
```javascript
jsPanel.create({
    contentSize: '3000 1500',
    position: {
        my: 'center',
        at: 'center',
        modify: function(values) {
            if(values.left < 0) {values.left = 0}
            if(values.top < 0) {values.top = 0}
            return values;
        }
    }
});
```
Nevertheless I think about adding minLeft, minTop, maxLeft and maxTop parameters to option.position for the ease of use??

**Comment by temuri416** [comment]
That solution works, but I think "min*" and "max*" properties are quite useful too.

**Comment by Flyer53** [comment]
Yep!
Not an official release yet, but you can download and test http://jspanel.de/downloads/jsPanel-3.11.0.zip
In this version option.position is extended with parameters **minLeft**, **minTop**, **maxLeft** and **maxTop**. Values have to be unitless numbers that will be interpreted as pixel values.
```javascript
$.jsPanel({
    contentSize: '3000 1500',
    position: {
        my: 'center',
        at: 'center',
        minLeft: 0,
        minTop: 0
    }
});
```
The same is implemented in a new alpha of v4.0.0 http://alpha.jspanel.de/downloads/jspanel-4-alpha.zip

**Comment by temuri416** [comment]
Works as expected in 3.11.0! 

Thanks :-)

**Comment by Flyer53** [comment]
Issue solved with v3.11.0 release (added option.position parameters)
